torchgeo/datasets/__init__.py

```diff
@@ -101,6 +101,7 @@
     BoundingBox,
     concat_samples,
     merge_samples,
+    rearrange_patches_to_sample,
     stack_samples,
     unbind_samples,
```

Comment:
One of these is not like the other...

Trying to think of a more consistent naming scheme here. What about `rearrange_samples`? Would that be too generic?

Reply:
Just wanted to mention that in other related PRs (Potsdam and GID-15), this function was named as `collate_patches_per_tile`. Perhaps that would be a better fit?

Reply:
The "one of these is not like the other" statement would still apply. We should either 1) change all of our existing collation functions to match this new naming style, or 2) change this function to match our existing collation function naming style.